### PR TITLE
Add java.util.Optional support

### PR DESCRIPTION
**Pull request by pratik-brex**

### CODE REVIEW

File: openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java [hunk 1]
```diff
@@ -640,13 +640,18 @@ private Map toTupleSubMap(Object[] a, String entryKey, String entryValue) {
...
+            boolean isNullable = false;
 
             // StreamingFileUpload implements Publisher, but it should be not considered as a Publisher in the spec file
             if (!type.isAssignable("io.micronaut.http.multipart.StreamingFileUpload") && isContainerType(type)) {
                 isPublisher = type.isAssignable(Publisher.class.getName()) && !type.isAssignable("reactor.core.publisher.Mono");
                 isObservable = type.isAssignable("io.reactivex.Observable") && !type.isAssignable("reactor.core.publisher.Mono");
                 type = type.getFirstTypeArgument().orElse(null);
+            } else if (type.isAssignable("java.util.Optional") || type.isAssignable("com.google.common.base")) {
+               isNullable = true;
+               type = type.getFirstTypeArgument().orElse(null);
```
Comment (alvarosanchez) [inline openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java:652]: This seems it can be only once after the if/else block

**Review by alvarosanchez — CHANGES_REQUESTED** [review]
Minor comment. Also, could you please add a test?

File: openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java (hunk 1 above)
Comment (pratik-brex) [reply openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java:652]: sorry do you mean that we should recursively resolve it? so `Optional<Optional<X>> ` resolves to `nullable: true` and type `X` ?

Comment (alvarosanchez) [reply openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java:652]: No, I just mean the line `type = type.getFirstTypeArgument().orElse(null);`, which is duplicated at the end of both branches of the if/else, could be left after the if/else block.

Comment (pratik-brex) [reply openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java:656]: it's an `else if` so it wouldn't be correct to move it outside the block.

Comment (alvarosanchez) [reply openapi/src/main/java/io/micronaut/openapi/visitor/AbstractOpenApiVisitor.java:642]: Ugh, you are right, ignore me

**Review by alvarosanchez — APPROVED** [review]

**Comment by alvarosanchez** [comment]
Thanks for your contribution!

**Pull request merged**